File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/CalendarViewModes.swift
//
//  CalendarViewModes.swift
//  Swift Examples
//
//  Copyright (c) 2014 Telerik. All rights reserved.
//

import Foundation

class CalendarViewModes: TKExamplesExampleViewController, TKCalendarDelegate {
    
    let calendarView = TKCalendar()
    
    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: NSBundle?) {
        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)

        self.addOption("Week view", action: selectWeekView)
        self.addOption("Month", action: selectMonth)
        self.addOption("Month Names", action: selectMonthNames)
        self.addOption("Year", action: selectYear)
        self.addOption("Year Numbers", action: selectYearNumbers)
        self.addOption("Flow", action: selectFlow)
    }

    required init(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.view.backgroundColor = UIColor.lightGrayColor()
        
        let date = NSDate()
        let calendar = NSCalendar(calendarIdentifier: NSCalendarIdentifierGregorian)!
        calendar.firstWeekday = 2
        let components = NSDateComponents()
        components.year = -10
        let minDate: NSDate = calendar.dateByAddingComponents(components, toDate: date, options: NSCalendarOptions(rawValue:0))!
        components.year = 10
        let maxDate: NSDate = calendar.dateByAddingComponents(components, toDate: date, options: NSCalendarOptions(rawValue:0))!
        
        self.calendarView.autoresizingMask = UIViewAutoresizing(rawValue:UIViewAutoresizing.FlexibleWidth.rawValue | UIViewAutoresizing.FlexibleHeight.rawValue)
        self.view.addSubview(self.calendarView)
        
        self.calendarView.delegate = self
        self.calendarView.viewMode = TKCalendarViewMode.Year
        self.calendarView.calendar = calendar
        self.calendarView.minDate = minDate
        self.calendarView.maxDate = maxDate
// >> view-modes-pinchzoom-swift
        self.calendarView.allowPinchZoom = true
// << view-modes-pinchzoom-swift
        self.calendarView.navigateToDate(date, animated: false)

// >> view-modes-presenter-swift
        let presenter = self.calendarView.presenter as! TKCalendarYearPresenter
        presenter.columns = 3
// << view-modes-presenter-swift
    }
    
    override func viewDidLayoutSubviews() {
        super.viewDidLayoutSubviews()
        if self.calendarView.viewMode == TKCalendarViewMode.Week {
            self.calendarView.frame = CGRectMake(0, self.view.bounds.origin.y, CGRectGetWidth(self.view.bounds), 100)
        }
        else {
            self.calendarView.frame = self.view.bounds
        }
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
//MARK: - Events
    
    func selectYear() {
// >> getting-started-viewmodeyear-swift
        self.calendarView.viewMode = TKCalendarViewMode.Year
// << getting-started-viewmodeyear-swift        
    }
    
    func selectMonth() {
// >> view-modes-month-swift
        self.calendarView.viewMode = TKCalendarViewMode.Month
// << view-modes-month-swift        
    }
    
    func selectMonthNames() {
// >> view-modes-monthnames-swift
        self.calendarView.viewMode = TKCalendarViewMode.MonthNames
// << view-modes-monthnames-swift
    }
    
    func selectYearNumbers() {
// >> view-modes-yearnumber-swift
        self.calendarView.viewMode = TKCalendarViewMode.YearNumbers
// << view-modes-yearnumber-swift
    }
    
    func selectFlow() {
// >> view-modes-flow-swift
        self.calendarView.viewMode = TKCalendarViewMode.Flow
// << view-modes-flow-swift        
    }
    
    func selectWeekView() {
// >> view-modes-week-swift
        self.calendarView.viewMode = TKCalendarViewMode.Week
// << view-modes-week-swift        
    }
    
//MARK: - TKCalendarDelegate

// >> view-modes-changeviewmode-swift
    func calendar(calendar: TKCalendar, didChangedViewModeFrom previousViewMode: TKCalendarViewMode, to viewMode: TKCalendarViewMode) {
        if viewMode == TKCalendarViewMode.Week || previousViewMode == TKCalendarViewMode.Week {
            self.view.setNeedsLayout()
        }
        
        self.selectedOption = UInt(viewMode.rawValue);
    }
}
// << view-modes-changeviewmode-swift

File: Examples/Examples/Calendar/CalendarViewModes.cs
using System;
using System.Drawing;

using Foundation;
using UIKit;
using CoreGraphics;

using TelerikUI;

namespace Examples
{
	[Register("CalendarViewModes")]
	public class CalendarViewModes : XamarinExampleViewController
	{
		CalendarDelegate calendarDelegate;

		public TKCalendar CalendarView {
			get;
			set;
		}

		public override void ViewDidLoad ()
		{
			this.AddOption ("Year", SelectYear);
			this.AddOption ("Month", SelectMonth);
			this.AddOption ("Month Names", SelectMonthNames);
			this.AddOption ("Year Numbers", SelectYearNumbers);
			this.AddOption ("Flow", SelectFlow);
			this.AddOption ("Week view", SelectWeekView);

			base.ViewDidLoad ();

			NSCalendar calendar = new NSCalendar (NSCalendarType.Gregorian);
			calendar.FirstWeekDay = 2;
			NSDate date = NSDate.Now;
			NSDateComponents components = new NSDateComponents ();
			components.Year = -10;
			NSDate minDate = calendar.DateByAddingComponents (components, date, (NSCalendarOptions)0);
			components.Year = 10;
			NSDate maxDate = calendar.DateByAddingComponents (components, date, (NSCalendarOptions)0);

			this.CalendarView = new TKCalendar (this.View.Bounds);
			this.CalendarView.AutoresizingMask = UIViewAutoresizing.FlexibleWidth | UIViewAutoresizing.FlexibleHeight;
			this.View.AddSubview (this.CalendarView);

			this.calendarDelegate = new CalendarDelegate(this);

			this.CalendarView.Delegate = calendarDelegate;
			this.CalendarView.ViewMode = TKCalendarViewMode.Year;
			this.CalendarView.Calendar = calendar;
			this.CalendarView.MinDate = minDate;
			this.CalendarView.MaxDate = maxDate;
			// >> view-modes-pinchzoom-cs
			this.CalendarView.AllowPinchZoom = false;
			// << view-modes-pinchzoom-cs
			this.CalendarView.NavigateToDate (date, false);

			// >> view-modes-presenter-cs
			TKCalendarYearPresenter presenter = (TKCalendarYearPresenter)this.CalendarView.Presenter;
			presenter.Columns = 3;
			// << view-modes-presenter-cs
		}

		public override void ViewDidLayoutSubviews ()
		{
			base.ViewDidLayoutSubviews ();
			if (this.CalendarView.ViewMode == TKCalendarViewMode.Week) {
				this.CalendarView.Frame = new CGRect (0, 0, this.View.Bounds.Width, 100);
			} else {
				this.CalendarView.Frame = this.View.Bounds;
			}
		}

		public void SelectYear ()
		{
			// >> getting-started-viewmodeyear-cs
			this.CalendarView.ViewMode = TKCalendarViewMode.Year;
			// << getting-started-viewmodeyear-cs
		}

		public void SelectMonth ()
		{
			// >> view-modes-month-cs
			this.CalendarView.ViewMode = TKCalendarViewMode.Month;
			// << view-modes-month-cs
		}

		public void SelectMonthNames ()
		{
			// >> view-modes-monthnames-cs
			this.CalendarView.ViewMode = TKCalendarViewMode.MonthNames;
			// << view-modes-monthnames-cs
		}

		public void SelectYearNumbers ()
		{
			// >> view-modes-yearnumber-cs
			this.CalendarView.ViewMode = TKCalendarViewMode.YearNumbers;
			// << view-modes-yearnumber-cs
		}

		public void SelectFlow ()
		{
			// >> view-modes-flow-cs
			this.CalendarView.ViewMode = TKCalendarViewMode.Flow;
			// << view-modes-flow-cs
		}

		public void SelectWeekView ()
		{
			// >> view-modes-week-cs
			this.CalendarView.ViewMode = TKCalendarViewMode.Week;
			// << view-modes-week-cs
		}

		class CalendarDelegate : TKCalendarDelegate
		{
			CalendarViewModes main;
			public CalendarDelegate (CalendarViewModes main)
			{
				this.main = main;
			}
			// >> view-modes-changeviewmode-cs
			public override void DidChangedViewModeFrom (TKCalendar calendar, TKCalendarViewMode previousViewMode, TKCalendarViewMode viewMode)
			{
				if (viewMode == TKCalendarViewMode.Week || previousViewMode == TKCalendarViewMode.Week) {
					this.main.View.SetNeedsLayout ();
				}
			}
			// << view-modes-changeviewmode-cs
		}
	}


}

